node/src/actors/chain_manager/actor.rs

```diff
@@ -246,8 +246,8 @@ impl ChainManager {
                 });
 
                 // Persist current chain state
-                act.last_chain_state = act.chain_state.clone();
-                act.persist_chain_state(ctx);
+                act.previous_chain_state = act.chain_state.clone();
+                act.persist_previous_chain_state(ctx);
```

Comment:
This is repeated twice, let's move the clone inside a method:

```suggestion
                act.persist_current_chain_state(ctx);
```

Reply:
Done, it's `self.advance_chain_state` now